src/Connection/Http2Connection.php

```diff
@@ -206,29 +206,35 @@ public function getProtocolVersions(): array
         return self::PROTOCOL_VERSIONS;
     }
 
-    public function getStream(Request $request): Stream
+    public function getStream(Request $request): Promise
     {
         if (!$this->initialized || $this->settingsDeferred !== null) {
             throw new \Error('The promise returned from ' . __CLASS__ . '::initialize() must resolve before using the connection');
         }
 
-        if ($this->remainingStreams <= 0) {
-            throw new \Error('All available streams have been used');
-        }
+        return call(function () use ($request) {
+            if ($this->streamId > 0 && empty($this->streams) && !yield $this->ping()) {
```

Comment:
This will ping multiple times concurrently if the condition is met.

Reply:
`ping()` only allows one un-acked ping to be in-flight at a time. Successive calls return the same promise.